File: graphql-test/src/main/java/com/junyss/graphqltest/api/equipment/repository/EquipmentRepository.java
package com.junyss.graphqltest.api.equipment.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import com.junyss.graphqltest.api.equipment.model.entity.Equipment;

public interface EquipmentRepository extends JpaRepository<Equipment, String> {

	@Query(value = "select equipment from Equipment equipment where equipment.equipmentId = :equipmentId")
	Optional<Equipment> findByEquipmentId(String equipmentId);
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/equipment/repository/EquipmentRepository.kt
package com.junyharangstudy.kotlingraphqltest.api.equipment.repository

import com.junyharangstudy.kotlingraphqltest.api.equipment.model.entity.Equipment
import org.springframework.data.jpa.repository.JpaRepository

interface EquipmentRepository: JpaRepository<Equipment, String>